### ISSUE TITLE
Embed the full app

### ISSUE THREAD

**Issue by ge0rg** [conversejs/converse.js#1297]
I'm trying to make a fullscreen "embedded" experience with the following HTML5:

```html
<body style="width: 100vw; height: 100vh; margin:0">
<div id="conversejs" style="padding:0">
<noscript><h1>This chat only works with JavaScript enabled!</h1></noscript>
<script>
converse.initialize({
...
});
</script>
</div>
</body>
```

However, it is slightly off to the right, having a minor padding to the left of everything, and scrolling a little bit. After joining a MUC, the scrolling happens in both directions.

Please properly document how to do a full-page embed :)

**Comment by ge0rg** [comment]
Apparently this is due to the offsets defined in `converse.css` for some screen sizes:

```css
  @media screen and (max-width: 480px) {
    #conversejs {
      margin: 0;
      right: 10px;
      left: 0px;
      bottom: 5px;
      } }
```

**Comment by ge0rg** [comment]
Another nitpick: mobile Chrome dynamically shows/hides the URL bar, but the viewport size seems to be calculated for a hidden URL bar, so the whole thing is always scrolling.

**Comment by ge0rg** [comment]
The following CSS works around a large part of the problems, but not the Chrome URL bar:

```css
#conversejs{margin:0;left:0;right:0;top:0;bottom:0}
```

**Comment by jcbrand** [comment]
I've split this ticket into two. Here we only address the ability to embed the full app inside an arbitrary DOM element.

**Comment by jcbrand** [comment]
The changes are backwards incompatible, so they'll only be included in the 5.0.0 release.

**Comment by joudinet** [comment]
Great, thanks.
I've noticed an issue when the screen max-width is below 480px. I'm not sure if you prefer I report this issue related to the embedded-full branch here or in a separate issue.
In the full screen view, it correctly shows either the controlbox or the chatroom but in the embedded view, it shows both the controlbox and the chatroom even though there is not enough space for the chatroom. See a screenshot of this issue below:
![embedded-full-small-width](https://user-images.githubusercontent.com/2621900/50834033-4d345780-1353-11e9-83d6-1a268ca905cd.png)
